tool/sdg/src/main/java/org/openscience/cdk/layout/StructureDiagramGenerator.java

```diff
@@ -297,12 +316,23 @@ public void generateCoordinates(Vector2d firstBondVector) throws CDKException {
         if (molecule.getAtomCount() == 1) {
             molecule.getAtom(0).setPoint2d(new Point2d(0, 0));
             return;
+        } else if (molecule.getBondCount() == 1) {
+            molecule.getAtom(0).setPoint2d(new Point2d(0, 0));
+            molecule.getAtom(1).setPoint2d(new Point2d(bondLength, 0));
+            return;
         }
-        if (!ConnectivityChecker.isConnected(molecule)) {
-            logger.debug("Molecule is not connected. Throwing exception.");
-            throw new CDKException(disconnectedMessage);
-        } else {
-            logger.debug("Molecule is connected.");
+
+        // intercept fragment molecules and lay them out in a grid
+        if (!isFragment) {
```

Comment:
Ah bad naming.. param could be named better 'isConnected'.. a fragment is connected... fragments are not..